faker/providers/file/__init__.py

```diff
@@ -269,16 +282,28 @@ def file_path(
         If ``absolute`` is ``True`` (default), the generated path starts with
         ``/`` and is absolute. Otherwise, the generated path is relative.
 
+        if ``file_system`` is set (default="linux"), the generated path uses
+        specified file system path standard, the list of valid file systems include:
+        ``'windows'``, ``'linux'``.
+
         :sample: size=10
         :sample: depth=3
         :sample: depth=5, category='video'
         :sample: depth=5, category='video', extension='abcdef'
+        :sample: depth=5, category='video', extension='abcdef', file_system='windows
         """
+        fs_rule = self.file_systems_path_rules.get(file_system_rule, None)
+        if not fs_rule:
+            raise TypeError("Specified file system is invalid.")
```

Comment:
Whhy do we declare the argument to be `Optional`, if then we eeror when it's `None`? Should we remove `Optional` from the type hint? The default value would still make it backward compatible.

Reply:
> Whhy do we declare the argument to be `Optional`, if then we eeror when it's `None`? Should we remove `Optional` from the type hint? The default value would still make it backward compatible.

for this my idea was something like user missple the ``windows`` word or set an invalid value.
if we don't support or missple the value we throw that error.
and ``Optional`` is for user to know that this param is not required.

Reply:
If we tell users that this defaults to `"linux"`, I don't think it owuld be necessary to have it declared as `Optional`